bakery/checkers/declared.go

```diff
@@ -76,33 +77,42 @@ func (c Declared) Check(_, arg string) error {
 // different values, the information is omitted from the map.
 // When the caveats are later checked, this will cause the
 // check to fail.
-func InferDeclared(ms macaroon.Slice) Declared {
-	var conflicts []string
-	info := make(Declared)
+func InferDeclared(ns *Namespace, ms macaroon.Slice) map[string]string {
+	var conditions []string
 	for _, m := range ms {
 		for _, cav := range m.Caveats() {
-			if cav.Location != "" {
-				continue
-			}
-			name, rest, err := ParseCaveat(string(cav.Id))
-			if err != nil {
-				continue
-			}
-			if name != CondDeclared {
-				continue
+			if cav.Location == "" {
+				conditions = append(conditions, string(cav.Id))
 			}
-			parts := strings.SplitN(rest, " ", 2)
-			if len(parts) != 2 {
-				continue
-			}
-			key, val := parts[0], parts[1]
-			if oldVal, ok := info[key]; ok && oldVal != val {
-				conflicts = append(conflicts, key)
-				continue
-			}
-			info[key] = val
 		}
 	}
+	return InferDeclaredFromConditions(ns, conditions)
+}
+
+// InferDeclaredFromConditions is like InferDeclared except that
```

Comment:
as InferDeclared is basically the same as InferDeclaredFromConditions with filtering let's make sure the comments for these two functions make sense when considered as a pair